databricks/koalas/series.py

```diff
@@ -5189,6 +5189,94 @@ def explode(self) -> "Series":
         internal = internal.copy(spark_frame=internal.spark_frame.drop(NATURAL_ORDER_COLUMN_NAME))
         return first_series(DataFrame(internal))
 
+    def argmax(self):
+        """
+        Return int position of the largest value in the Series.
+
+        If the maximum is achieved in multiple locations,
+        the first row position is returned.
+
+        Returns
+        -------
+        int
+            Row position of the maximum value.
+
+        Examples
+        --------
+        Consider dataset containing cereal calories
+
+        >>> s = ks.Series({'Corn Flakes': 100.0, 'Almond Delight': 110.0,
+        ...                'Cinnamon Toast Crunch': 120.0, 'Cocoa Puff': 110.0})
+        >>> s  # doctest: +SKIP
+        Corn Flakes              100.0
+        Almond Delight           110.0
+        Cinnamon Toast Crunch    120.0
+        Cocoa Puff               110.0
+        dtype: float64
+
+        >>> s.argmax()  # doctest: +SKIP
+        2
+        """
+        sdf = self._internal.spark_frame.select(self._internal.data_spark_columns)
+
+        # We should remember the natural sequence started from 0
+        seq_col_name = verify_temp_column_name(sdf, "__distributed_sequence_column__")
+        sdf = InternalFrame.attach_distributed_sequence_column(sdf, seq_col_name)
+
+        scol = scol_for(sdf, self._internal.data_spark_column_names[0])
+        max_value = sdf.select(F.max(scol)).head(1)[0][0]
+        if max_value is None:
+            raise ValueError("attempt to get argmax of an empty sequence")
+
+        # If the maximum is achieved in multiple locations, the first row position is returned.
+        max_value_position = sdf.filter(scol == max_value).head(1)[0][0]
+
+        return max_value_position
```

Comment:
How about:

```suggestion
        sdf = self._internal.spark_frame.select(self.spark.column, NATURAL_ORDER_COLUMN_NAME)

        max_value = sdf.select(
            F.max(scol_for(sdf, self._internal.data_spark_column_names[0])),
            F.first(NATURAL_ORDER_COLUMN_NAME),
        ).head()
        if max_value[1] is None:
            raise ValueError("attempt to get argmax of an empty sequence")
        elif max_value[0] is None:
            return -1

        # We should remember the natural sequence started from 0
        seq_col_name = verify_temp_column_name(sdf, "__distributed_sequence_column__")
        sdf = InternalFrame.attach_distributed_sequence_column(
            sdf.drop(NATURAL_ORDER_COLUMN_NAME), seq_col_name
        )

        # If the maximum is achieved in multiple locations, the first row position is returned.
        return sdf.filter(
            scol_for(sdf, self._internal.data_spark_column_names[0]) == max_value[0]
        ).head()[0]
```

to:

- support `ks.Series([np.nan]).argmax()`
- delay attaching the distributed sequence.

Reply:
Thanks, @ueshin !

Let me check and will add the test !

Reply:
Great!! accept this change.

Reply:
Oh, anyway I didn't notice that `Add suggestion to batch`. 😭